tgui/packages/tgui/interfaces/ShuttleBlueprints.tsx

```diff
@@ -0,0 +1,458 @@
+                  Connect To Existing Shuttle
+                </Button.Confirm>
+              </Stack.Item>
+            </Stack>
+          </Stack.Item>
+        </Stack>
+      </Stack.Item>
+    </Stack>
+  );
+};
+
+const ShuttleConfiguration = (props: ShuttleConfigurationProps) => {
+  const [name, setName] = useState<string>();
+  const [mergeArea, setMergeArea] = useState<AreaData>();
+  const {
+    visualizing,
+    onShuttleFrame,
+    onShuttle,
+    inDefaultArea,
+    currentArea,
+    neighboringAreas = new Map<string, string>(),
+    apcs = new Map<string, BooleanLike>(),
+    defaultApc,
+    apcInMergeRegion,
+    idle,
+    masterExists,
+  } = props;
+  const { name: currentAreaName = '', ref: currentAreaRef = '' } =
+    currentArea || {};
+  const { name: mergeAreaName = undefined, ref: mergeAreaRef = '' } =
+    mergeArea || {};
+  const { act } = useBackend<ShuttleBlueprintsData>();
+  const removalApcConflict = defaultApc && apcs?.[currentAreaRef];
+  const mergeApcConflict = apcInMergeRegion && apcs[mergeAreaRef];
+  return (
+    <Stack fill vertical align="center" justify="space-around">
+      <Stack.Item textAlign="center">
+        <h2>Current Area:</h2>
+        <h3>
+          {onShuttle
+            ? inDefaultArea
+              ? 'Default Area'
+              : currentAreaName
+            : 'Not on Shuttle'}
+        </h3>
+      </Stack.Item>
+      <Stack.Item>
+        <Input
+          fluid
+          placeholder="New Area Name"
+          onChange={(_, value) => setName(value)}
+        />
+        <Stack>
+          <Stack.Item>
+            <Button.Confirm
+              disabled={!(onShuttle && inDefaultArea)}
+              tooltip={
+                onShuttle
+                  ? inDefaultArea
+                    ? null
+                    : 'You can only create a new area from the default area.'
+                  : 'You must be on the linked shuttle to do this.'
+              }
+              onClick={() => act('createNewArea', { name: name })}
+            >
+              Create New Area
+            </Button.Confirm>
+          </Stack.Item>
+          <Stack.Item>
+            <Button.Confirm
+              disabled={!onShuttle || inDefaultArea}
+              tooltip={
+                onShuttle
+                  ? inDefaultArea
+                    ? 'You cannot rename the default area.'
+                    : null
+                  : 'You must be on the linked shuttle to do this.'
+              }
+              onClick={() => act('renameArea', { name: name })}
+            >
+              Rename Current Area
+            </Button.Confirm>
+          </Stack.Item>
+        </Stack>
+      </Stack.Item>
+      <Stack.Item width="100%">
+        <Stack fill justify="center">
+          <Stack.Item>
+            <Dropdown
+              placeholder="Select Area"
+              options={Object.entries(neighboringAreas).map(
+                ([ref, name]: [string, string]) => {
```

Comment:
Shouldn't need to type this... something's up

Reply:
I tried doing `neighboringAreas.map` and it gave me `type String has no call signatures`

Reply:
per @distributivgesetz this is probably because it's reading it as a map. map has its own `.entries()` function. change it to {} and it should be able to just `.map(([ref, name]) => { doThing(ref, name); })`

Reply:
if you were wondering, maps are just good for dynamic data, they're very rarely used here